jekyll/_cci2/resource-class-overview.adoc

```diff
@@ -0,0 +1,110 @@
+---
+contentTags: 
+  platform:
+  - Cloud
+  - Server v4.x
+  - Server v3.x
+  - Server v2.x
+---
+= Resource class overview
+:page-layout: classic-docs
+:page-liquid:
+:page-description: An overview of resource classes in CircleCI
+:icons: font
+:toc: macro
+:toc-title:
+
+Use resource classes to:
+
+* Specify compute resource size (CPU and RAM) for running a link:/docs/concepts/#jobs[job].
+* Optimize executors based on historical trends with resource class Insights.
+* Access the link:/docs/using-arm[Arm] execution environment.
+* Configure link:/docs/runner-concepts/#namespaces-and-resource-classes[self-hosted runners] for use in your `.circleci/config.yml` file.
+
+[#introduction]
+== Introduction 
+
+The `resource_class` configuration option has multiple uses. The most common usage is to specify CPU and RAM requirements for a job by configuring the `resource_class` key for an executor. 
+
+[source,yaml]
+----
+# .circleci/config.yml
+jobs:
+  my-job:
+    docker: # use the Docker executor
+      - image: cimg/base:2022.09 # specify a Docker image
+    resource_class: xlarge # specify a resource class
+    steps:
+      - run: echo "Hello World"
+----
+
+Executor resource classes are specified by size, for example, `small` or `medium`. Each size corresponds to specific CPU and RAM requirements. For a full list of available resource class options, see the link:/docs/configuration-reference/#resourceclass[configuration reference]. The link:https://circleci.com/product/features/resource-classes[resource class product page] provides information on credit usage per resource class, and resource class availability for each plan type.
```

Comment:
there's also this concept of "generation" for macOS resources which may also eventually make its way to other executors.  do we want to call that out here?

Reply:
Not familiar with this. Can you point me to an example?

Reply:
I've made a note of this and will follow up in another ticket